# Ejercicio 1.1

La tasa de bajas de nuestra cartera es del 2.5% cada mes, excepto los meses de Julio y Agosto que es un 1%, y los meses impares que sube al 3.5%.

Se pide, crear un función que introduciendo el número del mes  nos devuelva la tasa de baja mensual, mostrando un mensaje de error para meses que no correspondan con los números enteros del 1 al 12.

In [1]:
def tasa_bajas(mes):
    try:
        if mes > 12:
            raise ValueError

        if isinstance(mes, float):
            raise TypeError

        if mes == 6 or mes == 7:
            return .01
        elif mes % 2 != 0:
            return .035
        else:
            return 0.025

    except(TypeError):
        print('El mes debe ser un valor entero')

    except(ValueError):
        print('El valor no esta dentro del rango')

Mostramos la tasa de error para los 12 meses:

In [2]:
for i in range(1,13):
    print(i, tasa_bajas(i))

1 0.035
2 0.025
3 0.035
4 0.025
5 0.035
6 0.01
7 0.01
8 0.025
9 0.035
10 0.025
11 0.035
12 0.025


Chequeamos los mensajes de error:

In [3]:
print(tasa_bajas(5.5))
print(tasa_bajas('12'))
print(tasa_bajas(15))

El mes debe ser un valor entero
None
El mes debe ser un valor entero
None
El valor no esta dentro del rango
None


NOTA: Este ejemplo es ilustrativo para introducirse en la creación de funciones. Si se diera justo este caso, la mejor opción sería crear una lista **'tasa_bajas[0, 0.035, 0.025..]'** obteniendo el dato con un simple **'tasa_bajas[mes]'**
